src/shared/modules/connections/connectionsDuck.js

```diff
@@ -165,25 +163,17 @@ const removeConnectionHelper = (state, connectionId) => {
 }
 
 const mergeConnectionHelper = (state, connection) => {
-  const connectionId = connection.id
-  const connectionsById = { ...state.connectionsById }
-  const allConnectionIds = state.allConnectionIds
-  const index = allConnectionIds.indexOf(connectionId)
-  if (index >= 0) {
-    connectionsById[connectionId] = {
-      ...connectionsById[connectionId],
-      ...connection
-    }
-  } else {
-    connectionsById[connectionId] = {
-      ...connection
-    }
-    allConnectionIds.push(connectionId)
-  }
+  const { connectionsById, allConnectionIds } = state
+  const { id } = connection
   return {
     ...state,
-    allConnectionIds,
-    connectionsById
+    connectionsById: {
+      ...connectionsById,
+      [id]: { ...connectionsById[id], ...connection }
+    },
+    allConnectionIds: allConnectionIds.includes(id)
+      ? allConnectionIds
+      : [...allConnectionIds, id]
   }
```

Comment:
These method cleanup is think are cleaner, and in passing they also fixed a small bug that left a connection ID "undefined" in the connectionsById state